native/cocos/editor-support/spine-wasm/spine-skeleton-instance.cpp

```diff
@@ -9,8 +9,53 @@ SlotMesh globalMesh(nullptr, nullptr, 0, 0);
 extern "C" {
 extern void spineListenerCallBackFromJS();
 extern void spineTrackListenerCallback();
+AttachmentVertices* generateAttachmentVertices(spine::Attachment* attachment);
 }
+
 using namespace spine;
+extern HashMap<SkeletonData *, HashMap<Attachment *, AttachmentVertices *>*> spineAttachmentVerticesMap;
+extern HashMap<SkeletonData *, HashMap<spine::String, spine::String>*> spineTexturesMap;
+
+#define LOOP_UV_COORDS(tmp_var, uvs_var, loop_count) \
+    for (int _i = 0, _ii = 0; _i < (loop_count); ++_i, _ii += 2) { \
+        (tmp_var)[_i].texCoord.u = (uvs_var)[_ii]; \
+        (tmp_var)[_i].texCoord.v = (uvs_var)[_ii + 1]; \
+    }
+
+#ifdef CC_SPINE_VERSION_4_2
+    #define SET_SPINE_TEXTURE_ID(currMesh, attachment, attachmentVertices, texturesMap) \
+        do { \
+            auto* region = static_cast<AtlasRegion*>(attachment->getRegion()); \
+            if (region && region->page && (region->page->name != attachmentVertices->_textureName)) { \
+                currMesh.textureID = (*texturesMap)[region->page->name]; \
+            } else { \
+                currMesh.textureID = attachmentVertices->_textureUUID; \
+            } \
+        } while(0)
+
+    #define INIT_ATTACHMENT_VERTICES(attachmentVertices, attachment, slot, texturesMap, attachmentVerticesMap) \
+        do { \
+            if (!(attachmentVertices) && !(attachment)->getRegion()) { \
+                (attachment)->getSequence()->apply((slot), (attachment)); \
+                if ((attachment)->getRegion()) { \
+                    (attachmentVertices) = generateAttachmentVertices((attachment)); \
+                    if ((texturesMap)->containsKey((attachmentVertices)->_textureName)) { \
+                        (attachmentVertices)->_textureUUID = (*(texturesMap))[(attachmentVertices)->_textureName]; \
+                    } \
+                    (attachmentVerticesMap)->put((attachment), (attachmentVertices)); \
+                } \
+            } \
+        } while(0)
+#else
+    #define SET_SPINE_TEXTURE_ID(currMesh, attachment, attachmentVertices, texturesMap) \
+        do { \
+            currMesh.textureID = attachmentVertices->_textureUUID; \
+        } while(0)
+
+    #define INIT_ATTACHMENT_VERTICES(attachmentVertices, attachment, slot, texturesMap, attachmentVerticesMap) \
+        do { /* Empty for non-4.2 versions */ } while(0)
+#endif
```

Comment:
Shall we use template functions instead of preprocessor? It is easier to add break points.